test/controllers/user.controller.test.js

```diff
@@ -1152,183 +1213,199 @@ describe('User Controller', () => {
+              expect(res.body.success).to.be.eql(false);
+              done();
+              User.findByIdAndUpdate.restore();
+            });
+        });
       });
-    });
 
-    context('without token', () => {
-      it('returns error', (done) => {
-        request()
-          .put('/api/v1/user/editor/upgrade')
-          .send(userInfo)
-          .end((err, res) => {
-            expect(res).to.have.status(403);
-            expect(res.body.success).to.be.eql(false);
-            expect(res.body.message).to.be.eql('Token needed to access resources');
-            done();
+      context('with invalid data', () => {
+        context('when user id is empty', () => {
+          it('returns an error', (done) => {
+            request()
+              .put('/api/v1/user/editor/downgrade')
+              .set('authorization', adminToken)
+              .send({ userId: '' })
+              .end((err, res) => {
+                expect(res).to.have.status(412);
+                expect(res.body.success).to.be.eql(false);
+                expect(res.body.message).to.be.eql('Validation Error');
+                expect(res.body.error).to.be.eql('"User id" is not allowed to be empty');
+                done();
+              });
           });
+        });
       });
     });
+  });
 
-    context('when non admin token is used', () => {
-      it('returns forbidden error', (done) => {
-        request()
-          .put('/api/v1/user/editor/upgrade')
-          .set('authorization', userToken)
-          .send(userInfo)
-          .end((err, res) => {
-            expect(res).to.have.status(403);
-            expect(res.body.success).to.be.eql(false);
-            expect(res.body.message).to.be.eql('You are not permitted to perform this action');
-            done();
-          });
-      });
+  describe('Get all users', () => {
+    const dummyUsers = UserFactory.buildList(16);
+    beforeEach(async () => {
+      adminToken = await addUser(adminUser);
+      userToken = await addUser(regularUser);
+      await User.insertMany(dummyUsers);
     });
 
-    context('when findByIdAndUpdate returns an error', () => {
-      it('returns the error', (done) => {
-        sinon.stub(User, 'findByIdAndUpdate').throws(new Error('Type Error'));
-        request()
-          .put('/api/v1/user/editor/upgrade')
-          .set('authorization', adminToken)
-          .send(userInfo)
-          .end((err, res) => {
-            expect(res).to.have.status(400);
-            expect(res.body.success).to.be.eql(false);
-            done();
-            User.findByIdAndUpdate.restore();
-          });
+    describe('when users exist in db', () => {
+      context('when no parameters are passed', () => {
+        it('returns the default values', (done) => {
+          request()
+            .get('/api/v1/user/all')
+            .set('authorization', adminToken)
+            .end((err, res) => {
+              expectsPaginationToReturnTheRightValues(res, {
```

Comment:
Lets use a default value for this and other tests to make the test assertion more readable:

```suggestion
              expectsPaginationToReturnTheRightValues(res, expectedPaginationResult);
```